ReactAndroid/src/main/java/com/facebook/react/modules/location/LocationModule.java

```diff
@@ -188,18 +190,34 @@ public void stopObserving() {
     mWatchedProvider = null;
   }
 
+
+  private boolean isProviderEnabled(LocationManager locationManager, String provider) {
+    try {
+      // Before API version LOLLIPOP, this method would throw SecurityException
+      // if the location permissions were not sufficient to use the specified provider.
+      return locationManager.isProviderEnabled(provider);
+    } catch (SecurityException e) {
+      return false;
+    }
+  }
+
   @Nullable
-  private static String getValidProvider(LocationManager locationManager, boolean highAccuracy) {
+  private String getValidProvider(LocationManager locationManager, boolean highAccuracy) {
     String provider =
         highAccuracy ? LocationManager.GPS_PROVIDER : LocationManager.NETWORK_PROVIDER;
-    if (!locationManager.isProviderEnabled(provider)) {
+    if (!isProviderEnabled(locationManager, provider)) {
       provider = provider.equals(LocationManager.GPS_PROVIDER)
           ? LocationManager.NETWORK_PROVIDER
           : LocationManager.GPS_PROVIDER;
-      if (!locationManager.isProviderEnabled(provider)) {
+      if (!isProviderEnabled(locationManager, provider)) {
         return null;
       }
     }
+    // If it's an enabled provider, but we don't have permissions, ignore it
+    int finePermission = ContextCompat.checkSelfPermission(getReactApplicationContext(), android.Manifest.permission.ACCESS_FINE_LOCATION);
```

Comment:
Not sure exactly what you're asking, but maybe this answers it?

My app was released prior to RN, and I used COARSE but not FINE permissions. I am not ready to block my app upgrades on "new FINE permission" that RN sometimes triggers accidentally, so wanted to avoid that.

This check is to say "If they are trying to use the GPS, but they don't have a FINE permission enabled, then pretend like we don't have a GPS so we don't trigger the permissions errors that crash the app" (These are not returned through onError, but propagate as a native Java exception).